packages/salesforce-adapter/src/filters/layouts.ts

```diff
@@ -77,12 +78,51 @@ const fixLayoutPath = (
   layout.path = [...objectPath.slice(0, -1), layout.elemID.typeName, layout.elemID.name]
 }
 
+type LayoutItem = {
+  field: string | ReferenceExpression
+}
+
+const fixLayoutItemField = (
+  layoutItem: LayoutItem, layoutObj: ObjectType, layoutObjFields: Field[]
+): void => {
+  if (!layoutItem) return
+  const findField = (fieldName: string, fields: Field[]): Field | undefined =>
+    fields.find(field => field.elemID.name === fieldName)
```

Comment:
`field.elemID` is the salto ID
`layoutItem.field` comes from salesforce and is the salesforce ID.
comparing these two is not the correct thing to do, we should compare to the salesforce ID

```suggestion
    fields.find(field => apiName(field, true) === fieldName)
```

Reply:
Fixed